message_set.go

```diff
@@ -48,9 +48,11 @@ func (msb *MessageBlock) decode(pd packetDecoder) (err error) {
 type MessageSet struct {
 	PartialTrailingMessage bool // whether the set on the wire contained an incomplete trailing MessageBlock
 	Messages               []*MessageBlock
+	UnderlyingMessageCount int // in case of compressed messages wraps in a single fake "message"
```

Comment:
Oh I need to learn to read, it's for compressed messages; I'm still not sure what purposes it servers when you could just ask for `len` of the compressed message-set's messages?